pkg/cmd/proxy/health/exec.go

```diff
@@ -103,10 +103,8 @@ func (o *Options) run(streams genericclioptions.IOStreams) error {
 		return errors.Wrapf(err, "failed initializing proxy api client")
 	}
 
-	// TODO: fix this deprecated field AddOnConfiguration
-	// nolint:staticcheck
 	proxyConfig, err := proxyClient.ProxyV1alpha1().ManagedProxyConfigurations().
-		Get(context.TODO(), clusterAddon.Spec.AddOnConfiguration.CRName, metav1.GetOptions{})
+		Get(context.TODO(), "cluster-proxy", metav1.GetOptions{})
```

Comment:
The name of managedproxyconfiguration is always `cluster-proxy` if we deploy addon via clusteradm: https://github.com/open-cluster-management-io/cluster-proxy/blob/2a197b382114b674b76a362146a7259aed164424/charts/cluster-proxy/templates/managedproxyconfiguration.yaml#L4